File: 351002/Yakhnovets/rest-task/src/mark/controller/mark.controller.ts
import {
  Body,
  Controller,
  Delete,
  Get,
  HttpCode,
  Param,
  ParseIntPipe,
  Post,
  Put,
} from '@nestjs/common';
import { MarkService } from '../service/mark.service';
import { MarkRequestTo } from '../dto/mark-request.to';
import { MarkResponseTo } from '../dto/mark-response.to';

@Controller('api/v1.0/marks')
export class MarkController {
  constructor(private readonly service: MarkService) {}

  @Get()
  getAll(): Promise<MarkResponseTo[]> {
    return this.service.getAll();
  }

  @Get(':id')
  getById(@Param('id', ParseIntPipe) id: number): Promise<MarkResponseTo> {
    return this.service.getById(id);
  }

  @Post()
  create(@Body() dto: MarkRequestTo): Promise<MarkResponseTo> {
    return this.service.create(dto);
  }

  @Put(':id')
  update(
    @Param('id', ParseIntPipe) id: number,
    @Body() dto: MarkRequestTo,
  ): Promise<MarkResponseTo> {
    return this.service.update(id, dto);
  }

  @Delete(':id')
  @HttpCode(204)
  delete(@Param('id', ParseIntPipe) id: number): Promise<void> {
    return this.service.delete(id);
  }
}

File: 351004/Hasiuk/Labs/src/main/java/com/example/Labs/controller/MarkController.java
package com.example.Labs.controller;

import com.example.Labs.dto.request.MarkRequestTo;
import com.example.Labs.dto.response.MarkResponseTo;
import com.example.Labs.service.MarkService;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.data.domain.Pageable;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.*;
import java.util.List;

@RestController
@RequestMapping("/api/v1.0/marks")
@RequiredArgsConstructor
public class MarkController {
    private final MarkService service;

    @PostMapping
    @ResponseStatus(HttpStatus.CREATED)
    public MarkResponseTo create(@Valid @RequestBody MarkRequestTo request) {
        return service.create(request);
    }

    @GetMapping
    public List<MarkResponseTo> getAll(Pageable pageable) {
        return service.getAll(pageable).getContent();
    }

    @GetMapping("/{id}")
    public MarkResponseTo getById(@PathVariable Long id) {
        return service.getById(id);
    }

    @PutMapping("/{id}")
    public MarkResponseTo update(@PathVariable Long id, @Valid @RequestBody MarkRequestTo request) {
        return service.update(id, request);
    }

    @DeleteMapping("/{id}")
    @ResponseStatus(HttpStatus.NO_CONTENT)
    public void delete(@PathVariable Long id) {
        service.delete(id);
    }
}